### PR TITLE
Add pLHE relvals to standard relval campaign

### PR DESCRIPTION
**Pull request by chayanit**
#### PR description:

According to JIRA (https://its.cern.ch/jira/browse/PDMVRELVALS-27) and discussed in preMccM meeting, we would like to include one pLHE relval to test the whole machinery in new releases, getting ready for UL, starting from 10_6_0

#### PR validation:

1. Local test has been done under IB CMSSW_11_0_X_2019-05-21-2300 by running 5 events for every new added workflows (1307,1307.17,1307.18 and 25215,25215.17,25215.18) with command line
e.g.
runTheMatrix.py --what standard -l 1370.0 -t 4 --command='-n 5' >& run5evt.log &
2. Also tested by real injection on 10_6_0 and everything ran successfully for noPU, while PU relvals are injected and under production

https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?campaign=CMSSW_10_6_0__fullsim_noPU_pLHE-1557911007
https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?campaign=CMSSW_10_6_0__fullsim_PU_pLHE-1558498719

#### if this PR is a backport please specify the original PR: 

This PR is needed for a backport to 10_6_X

### CODE REVIEW

**Comment by zhenhu** [comment]
please test workflow 25215,25215.17,25215.18,1370,1370.17,1370.18

**Comment by prebello** [comment]
@fabiocos  PdmV has requested @chayanit  to update all GENSIM strings (reproduced in 10-6-0) pushing a new commit in this PR. 
Is it ok for you or do you prefer that a new PR is made for this specific purpose?

**Comment by kpedro88** [comment]
@prebello I would suggest a separate PR, otherwise we mix two issues (comparisons will change from updated GEN-SIM strings, but adding a new pLHE relval should have no impact on comparisons for other workflows)

**Comment by prebello** [comment]
@kpedro88  ok then. So, the part of GENSIM strings in this PR (added because of pLHE relvals) can be maintained or should be moved to the new PR too?

**Comment by kpedro88** [comment]
@prebello ah, I didn't realize the new relvals already needed updated GEN-SIM strings.

The best approach would be to make a PR with all the updated GEN-SIM strings, test and merge it, then re-test this PR on top of that.

**Comment by prebello** [comment]
Hi @chayanit could you please check if the file open error in your relvals is related to the very recent change mentioned at
https://hypernews.cern.ch/HyperNews/CMS/get/wmDevelopment/805.html ?

**Comment by prebello** [comment]
Hi @chayanit could you please check if the file open error is related to the recent change mentioned at
https://hypernews.cern.ch/HyperNews/CMS/get/wmDevelopment/805.html ?
Failed to open file at URL root://eoscms.cern.ch:1094//eos/cms/store/user/cmsbuild/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz

**Comment by chayanit** [comment]
Hi @prebello,
This seems to be different issue. The path to EOS is somehow not correct where I marked bold shouldn't be there
root://eoscms.cern.ch:1094//eos/cms/**store/user/cmsbuild**/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz

**Comment by fabiocos** [comment]
@kpedro88 indeed the change of the GEN-SIM input would change the comparison result, but this should be irrelevant for the test of this PR in my view since:

- new workflows are added, and there is no previous reference to compare with;

- step1LHENormal and step1GENNormal are so far unused, for what I can see, so their redefinition should not impact any existing workflow (this would be the main possible problem in my view in mixing the two different updates). Do you agree?

@prebello @chayanit My main concern is whether all the new input files are available, I would like to avoid having the IB broken for a long period with plenty of DAS errors...

Furthermore, I think it would be also useful to add a test with multiple input lumi sections, see #26348

**Comment by fabiocos** [comment]
@prebello @chayanit the hn you mention discusses about WMAgent, @smuzaffar I do not see how that should enter into the bot operation, that is based just on runTheMatrix.py

**Comment by fabiocos** [comment]
please test workflow 25215,25215.17,25215.18,1370,1370.17,1370.18

**Comment by kpedro88** [comment]
@fabiocos it's not clear to me if the GEN-SIM strings updated in this PR only affect the new workflows, or could also affect existing workflows.

**Comment by chayanit** [comment]
Hi @fabiocos , @kpedro88 , before we continue, there is issue here about the access to EOS as I mentioned above that the path seems to not correct with additional **store/user/cmsbuild/**. 
root://eoscms.cern.ch:1094//eos/cms/**store/user/cmsbuild/**store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz
PS. just to answer Fabio's question : I can confirm that all new input files are available as we have already produced in the validation campaigns of 10_6_0, just reported in PPD meeting today.

**Comment by prebello** [comment]
@chayanit I propose, as @kpedro88 has pointed to be the best procedure, to make a new PR only for GENSIM strings update to avoid delays for the next release. 
Then we can check the error with pLHE relvals in parallel.

**Comment by fabiocos** [comment]
@kpedro88 @prebello @chayanit fine for me splitting the PR in two, but please point me where NOW are the modified sentences used within PyRelVal

**Comment by chayanit** [comment]
@fabiocos , @kpedro88 , @prebello , I made another PR for only strings update #26953. 
The modified sentences used for pLHE (private LHE) are https://github.com/cms-sw/cmssw/blob/6a7976d6f9825ed43692fd54915debd1ed6681c7/Configuration/PyReleaseValidation/python/relval_steps.py#L1425-L1470

**Comment by fabiocos** [comment]
@chayanit thank you for splitting the PR. Concerning my question I do see where in your PR the new steps are used, what I miss is where they are used before your PR. If you add something new that is not used before you cannot modify the previous behaviour, that is my point.

Anyway let's see the outcome of the new PR test.

**Comment by chayanit** [comment]
@fabiocos, ah sorry i didn't understand your question. Indeed, this is new added feature but made use of existing packages (step1LHENormal and step1GENNormal) which never used before and it currently failed to access file on EOS. Could you guide me what needs to be done when adding new feature? Thanks in advance.

**Comment by fabiocos** [comment]
please test workflow 25215,25215.17,25215.18,1370,1370.17,1370.18

**Comment by fabiocos** [comment]
@smuzaffar there is a problem in the way
```
# Input source
process.source = cms.Source("LHESource",
    fileNames = cms.untracked.vstring('/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz')
)
```
is translated in the test into a query
```
Note: this tool is DEPRECATED, use xrdfs instead.
05-Jun-2019 14:28:16 CEST  Initiating request to open LHE file root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz
%MSG-w XrdAdaptorInternal:  LHESource:source@sourceConstruction  05-Jun-2019 14:28:17 CEST pre-events
Failed to open file at URL root://eoscms.cern.ch:1094//eos/cms/store/user/cmsbuild/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz.
%MSG
%MSG-w XrdAdaptorInternal:  LHESource:source@sourceConstruction  05-Jun-2019 14:28:17 CEST pre-events
Failed to open file at URL root://eoscms.cern.ch:1094//eos/cms/store/user/cmsbuild/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz?tried=.
%MSG
----- Begin Fatal Exception 05-Jun-2019 14:28:17 CEST-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type LHESource
   [2] Calling StorageFactory::open()
   [3] Calling XrdFile::open()
Exception Message:
Failed to open the file 'root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz'
   Additional Info:
      [a] XrdCl::File::Open(name='root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz', flags=0x10, permissions=0660) => error '[ERROR] Server responded with an error: [3011] Unable to open file /eos/cms/store/user/cmsbuild/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz; No such file or directory
' (errno=3011, code=400). No additional data servers were found.
      [b] Last URL tried: root://eoscms.cern.ch:1094//eos/cms/store/user/cmsbuild/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz?tried=
      [c] Problematic data server: eoscms.cern.ch:1094
      [d] Disabled source: eoscms.cern.ch:1094
----- End Fatal Exception -------------------------------------------------
```

Running in a standalone way I manage to succeed:

```
Note: this tool is DEPRECATED, use xrdfs instead.
05-Jun-2019 14:46:15 CEST  Initiating request to open LHE file root://eoscms.cern.ch//eos/cms/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz
05-Jun-2019 14:46:17 CEST  Successfully opened LHE file root://eoscms.cern.ch//eos/cms/store/lhe/18334/HJ_nnlops_M125_nojhugen.lhe.xz
Begin processing the 1st record. Run 1, Event 1, LumiSection 1 on stream 0 at 05-Jun-2019 14:46:19.116 CEST
```

**Comment by fabiocos** [comment]
@smuzaffar this is likely due to the bot assuming ```/store/user/cmsbuild``` as the area where to look at for input files. The simplest solution to get this working would be to add the lhe area there with the few input files needed:
```
15:53 cmsdev25 1031> eos ls -l /store/user/cmsbuild/store/
drwxr-xr-+   1 cmsbuild zh         2160047665 Oct 30  2018 backfill
drwxr-xr-+   1 cmsbuild zh       1119278436337 Oct 24  2016 data
drwxr-xr-+   1 cmsbuild zh         6578997585 Mar 28  2018 express
drwxr-xr-+   1 cmsbuild zh          293000648 Oct 24  2016 generator
drwxr-xr-+   1 cmsbuild zh        46588672770 Oct 24  2016 hidata
drwxr-xr-+   1 cmsbuild zh       4025022381732 May 29 10:58 relval
drwxr-xr-+   1 cmsbuild zh         2399352099 Feb 18  2017 user
```

**Comment by smuzaffar** [comment]
I do not know why it is not going to fallback xrootd server. Normally whne a file is not found via root://eoscms.cern.ch:1094//eos/cms/store/user/cmsbuild/store/ then cmsRun falls back to global XROOT redirector and use AAA to get files. So no idea why lhe files are not using the fallback xrootd ?

Anyway, I have created a symlink for lhe files so hopefully this should avoid this error.

[cmsbuild@lxplus730 store]$ ls -l /eos/cms/store/user/cmsbuild/store/lhe
lrwxrwxrwx. 1 cmsbuild zh 18 Jun  5 15:57 /eos/cms/store/user/cmsbuild/store/lhe -> /eos/cms/store/lhe

**Comment by fabiocos** [comment]
please test workflow 25215,25215.17,25215.18,1370,1370.17,1370.18

@smuzaffar thanks, let me try again

**Comment by fabiocos** [comment]
@smuzaffar these are ascii lhe files, they are not in the standard DAS/DBS catalogue to my knowledge, I doubt we distribute them, so I don't think xrootd and AAA are expected to work in this case

**Comment by chayanit** [comment]
Hi @fabiocos, do I understand correctly that the access to EOS is now solved? There is still small issue about GEN-SIM strings that caused failures and we are trying to figure out. 
FYI. @zhenhu, @prebello

**Comment by fabiocos** [comment]
@chayanit yes, the eos issue for LHE files is now solved. What is the remaining problem? Is the PR ready to be tested again or do you need more work on it? BTW, this now partly overlaps with #26953

**Comment by chayanit** [comment]
Hi @fabiocos, thanks for your prompt reply. Actually this PR is ready for another test and hopefully be merged soon. 
>>BTW, this now partly overlaps with #26953
Indeed. as you may remember, we did not want to delay the updated GS string and made another PR #26953 but then it failed due to missing 57 relvals and we asked you to revert back from merge. Now we injected those missing relvals and want to merge together with this PR. Is that possible because these 2 PRs have some overlap on GS string.
Cheers,
Chayanit

**Comment by fabiocos** [comment]
please test workflow 25215,25215.17,25215.18,1370,1370.17,1370.18

**Comment by fabiocos** [comment]
@chayanit @prebello @zhenhu here is the output of my test about the availability of the samples after the possible integration of this PR: all of them are available in some site but for

**Comment by zhenhu** [comment]
please test workflow 25215,25215.17,25215.18,1370,1370.17,1370.18

**Comment by zhenhu** [comment]
Hi @fabiocos ,
These 3 datasets are indeed missing and we will produce them soon: 
dataset=/RelValNuGun/CMSSW_10_6_0-106X_mc2017_realistic_v3-v1/GEN-SIM
dataset=/RelValQQH1352T_13/CMSSW_10_6_0-106X_mc2017_realistic_v3-v1/GEN-SIM
dataset=/RelValSMS-T1tttt_mGl-1500_mLSP-100_13/CMSSW_10_6_0-106X_mc2017_realistic_v3-v1/GEN-SIM

This one have been fixed in the last commit: 
dataset=/RelValQCD_Pt_600_800_13/CMSSW_10_6_0-106X_upgrade2021_realistic_v4_LowSigmaZ-v1/GEN-SIM

This one has the v2 issue:
/RelValH125GGgluonfusion_13/CMSSW_10_6_0-106X_upgrade2018_realistic_v4-v1/GEN-SIM
Since it's a premix workflow and the v1 is hardcoded, there is no simple solution. But the issue will be fixed after we reproduced the GENSIM in the next release

**Comment by chayanit** [comment]
> @chayanit @prebello @zhenhu here is the output of my test about the availability of the samples after the possible integration of this PR: all of them are available in some site but for
> 
> dataset=/RelValH125GGgluonfusion_13/CMSSW_10_6_0-106X_upgrade2018_realistic_v4-v1/GEN-SIM
> dataset=/RelValNuGun/CMSSW_10_6_0-106X_mc2017_realistic_v3-v1/GEN-SIM
> dataset=/RelValQCD_Pt_600_800_13/CMSSW_10_6_0-106X_upgrade2021_realistic_v4_LowSigmaZ-v1/GEN-SIM
> dataset=/RelValQQH1352T_13/CMSSW_10_6_0-106X_mc2017_realistic_v3-v1/GEN-SIM
> dataset=/RelValSMS-T1tttt_mGl-1500_mLSP-100_13/CMSSW_10_6_0-106X_mc2017_realistic_v3-v1/GEN-SIM
> 
> So if we integrate this PR we will have test workflows based on these 5 samples failing until they do not appear in DAS/DBS. What is the ETA for them?

Hi @fabiocos , i have injected 4 of them and they should be available within 2-3 days, hopefully. As Zhen mentioned, the first one can simply be fixed after we reproduce the GENSIM and update strings in the next release (11_0_0_pre1).

**Comment by fabiocos** [comment]
@chayanit @zhenhu just to understand, are you planning another update of input files after 11_0_0_pre1 GEN-SIM is made? Are you waiting to update this PR for that?

**Comment by chayanit** [comment]
Hi @fabiocos, from my understanding, we will not wait for another update of 11-0-0-pre1 GENSIM strings. That will come in another PR. Now, I'm just waiting the missing relvals to finish. Do you agree?
https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?campaign=CMSSW_10_6_0__fullsim_noPU_adhoc2017_v2-1560145243
https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?campaign=CMSSW_10_6_0__fullsim_noPU_adhoc2021_v2-1560144739

**Comment by fabiocos** [comment]
@chayanit so far only these two samples are missing:

dataset=/RelValH125GGgluonfusion_13/CMSSW_10_6_0-106X_upgrade2018_realistic_v4-v1/GEN-SIM
dataset=/RelValTTbar_13/CMSSW_9_4_0-94X_mc2017_realistic_v10-v1/GEN-SIM-RECO

**Comment by zhenhu** [comment]
Hi @fabiocos , we will try to reproduce H125GGgluonfusion_13. 
But for the other one, why is it GEN-SIM-RECO?

**Comment by zhenhu** [comment]
+1

**Comment by fabiocos** [comment]
@zhenhu 

```
16:31 cmsdev25 4657> find . -name cmdLog -exec grep -n -H dataset=/RelValTTbar_13/CMSSW_9_4_0-94X_mc2017_realistic_v10-v1/GEN-SIM-RECO {} \;
./1325.51_TTbar_13_94XreminiaodINPUT+TTbar_13_94XreminiaodINPUT+REMINIAOD_mc2017+HARVESTUP17_REMINIAOD_mc2017/cmdLog:3: dasgoclient --limit 0 --query 'file dataset=/RelValTTbar_13/CMSSW_9_4_0-94X_mc2017_realistic_v10-v1/GEN-SIM-RECO site=T2_CH_CERN' | sort -u > step1_dasquery.log  2>&1
```

**Comment by chayanit** [comment]
@fabiocos , I also injected this missing TTbar reminiaod one. It should be available soon and hopefully this PR can be merged. Thank you.

**Comment by fabiocos** [comment]
@kpedro88 I think at this point this could enter before 11_0_0_pre2, do you agree?

**Comment by kpedro88** [comment]
The original plan was to have separate PRs for updating strings and adding new relvals. I'm not sure why this plan has been abandoned. I would prefer to be able to test the two changes separately.

**Comment by zhenhu** [comment]
Hi @kpedro88 , the pLHE workflow also require new strings. So it's a bit tricky to separate them.

**Comment by chayanit** [comment]
Hi @kpedro88, there were two separate PRs as you asked #26953. 

>The best approach would be to make a PR with all the updated GEN-SIM strings, test and merge it, then re-test this PR on top of that.

The GEN-SIM strings PR was actually merged but because of missing non-relevant relvals, so we asked you to revert back to not delay things and we submitted those missing ones. So in principle, we just worked on top of that PR for missing relvals.

**Comment by kpedro88** [comment]
Yes, since that PR was reverted, it should be remade (assuming now all the necessary relvals are available). Then this one can go on top of that one.

**Comment by chayanit** [comment]
but there are some part of GENSIM duplicated. Maybe I don't get your point but why can't they go together?

**Comment by kpedro88** [comment]
A PR entitled "add pLHE relvals..." should not generate 7000 comparison differences.

I understand that some of these GEN-SIM strings are *also* needed for the new relvals. My point is that the general update of the strings should happen first, and then, when the new strings are available, these relvals can be added. Doing it all at once just introduces confusion.

**Comment by fabiocos** [comment]
@chayanit the situation looks quite clear to me, in order to avoid too many discussions please repropose an updated version of #26953 , and then rebase this PR on that

**Comment by chayanit** [comment]
@kpedro88 , @fabiocos , ok i recreated another PR only for GENSIM strings #27206

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
@kpedro88 as expected the GEN-SIM input update and other developments in this PR factorise, I believe we may move forward at this point, unless you have further remarks

**Comment by fabiocos** [comment]
+1
